### PR TITLE
fix(ai,nodes): make Whisper decode params, node config and model identity actually reach their targets

### LINKED ISSUE #1809: Whisper: beam_size, vad_filter and vad_parameters are sent per request but never reach the decoder
## Summary

`Whisper.transcribe()` accepts `beam_size`, `vad_filter` and `vad_parameters`, and `_transcribe_remote()` sends all three with every request — but `WhisperLoader.inference()` hardcodes them instead of declaring them as parameters, so in server mode they are silently discarded and every transcription decodes at the built-in defaults.

This is the same class of bug as #1093 / #1094 / #1095, which are now fixed: a per-call parameter that never reaches the model. It was left out of scope on #1751 deliberately, since that PR was about `language`.

## The mismatch

`whisper.py:643` — the facade takes them:

```python
def transcribe(
    self,
    audio: bytes,
    beam_size: int = 5,
    vad_filter: bool = True,
    vad_parameters: Optional[Dict[str, Any]] = None,
    language: Optional[str] = None,
    **kwargs,
) -> Dict[str, Any]:
```

`whisper.py:733` — and puts them on the wire:

```python
'beam_size': beam_size,
'vad_filter': vad_filter,
'vad_parameters': vad_parameters,
```

`whisper.py:336` — but `inference()` declares only `language`:

```python
def inference(
    model, preprocessed, metadata=None, stream=None,
    language: Optional[str] = None,
) -> List[Dict[str, Any]]:
```

`whisper.py:397` — and hardcodes the rest:

```python
segments_gen, info = whisper_model.transcribe(
    audio,
    beam_size=5,
    vad_filter=True,
    vad_parameters={...},
)
```

## Why they are dropped

The saas model server now forwards per-request parameters into the loader stages (rocketride-ai/rocketride-saas#466), but it narrows them to what each stage's signature declares — no loader's `inference()` accepts `**kwargs`, so an unfiltered splat would raise `TypeError` the moment a parameter reached a loader that did not know it.

`language` is declared, so it arrives. `beam_size`, `vad_filter` and `vad_parameters` are not, so the filter drops them and the hardcoded values win. Local mode is unaffected — `_transcribe_local()` calls `inference()` directly and hits the same hardcoded values, so both modes ignore them, just for different reasons.

## Impact

A caller raising `beam_size` for accuracy, or disabling `vad_filter` for audio where VAD hurts, gets no effect and no error. `vad_parameters` is the most visible: it is `None` by default in the facade, so a caller who tunes silence thresholds is silently overridden by the hardcoded dict.

Nothing in-tree passes them today — `nodes/src/nodes/audio_transcribe` does not — so this is latent rather than actively firing, the same shape as #1095.

## Suggested fix

Declare them on `inference()` and use them, mirroring what #1751 did for `language`:

```python
def inference(
    model, preprocessed, metadata=None, stream=None,
    language: Optional[str] = None,
    beam_size: int = 5,
    vad_filter: bool = True,
    vad_parameters: Optional[Dict[str, Any]] = None,
) -> List[Dict[str, Any]]:
```

then pass them through to `whisper_model.transcribe()`, keeping the current values as defaults so behaviour is unchanged for callers who do not set them.

Worth confirming while doing it: these are decode-time parameters, so they must not go into `loader_options` — that would split model identity and load a duplicate copy of the weights per beam size, which is exactly what #1093/#1094/#1095 removed.

## Acceptance

- `Whisper.transcribe(audio, beam_size=1)` and `beam_size=10` produce different output in **server** mode, not just local
- A test asserting the values reach `whisper_model.transcribe()` — `test_whisper_identity.py` already has a `_FakeWhisperModel` that records decode kwargs
- Model identity unchanged by any of the three

### PR DESCRIPTION
**Pull request by asclearuc**
Closes #1809.

`Whisper.transcribe()` accepted `beam_size`, `vad_filter` and `vad_parameters` and put all three on the wire, but `WhisperLoader.inference()` declared only `language` and hardcoded the rest — so the model server's signature narrowing dropped them, and `_transcribe_local()` never forwarded them either. Both modes ignored them, for different reasons.

Fixing that surfaced five more instances of the same defect in this node: a parameter declared in one place and not read in the other. All six are fixed here, because each one alone leaves a knob that silently does nothing.

| where | what was wrong |
| --- | --- |
| `WhisperLoader.inference()` | `beam_size` / `vad_filter` / `vad_parameters` not declared |
| node `services.json` | `silence_threshold`, `min_seconds`, `max_seconds`, `vad_level` declared, read by nothing |
| `preconfig.profiles` | set `mode`, the node reads `model` |
| `inference()` | `word_timestamps` never passed, so the `words` block was dead code |
| `IInstance` | built `Transcribe()` without the kwarg it accepts |
| `WhisperLoader.load()` | genuine loader options changed identity but never reached `WhisperModel()` |

Reasoning, measurements and the full findings are on #1809.

## For the reviewer

**Every profile was loading `base`.** `preconfig.profiles` set `mode` while `IGlobal` reads `config.get('model', 'base')`, with no translation anywhere. Picking `large-v3` gave you `base` — including `nodes:test-full`, which advertises five model sizes and ran `base` five times. This is the one change with user-visible behaviour: anyone on a non-`default` profile now gets the model they asked for, with the matching speed and memory cost.

**Four config fields are removed, not deprecated.** Nothing validates node config keys — the base `validateConfig` is a bare `pass`, the contract test checks only `title`/`protocol`/`node_type`, `getNodeConfig` does not filter, and saas does not either — so configs still carrying them keep working. Removing declared fields is established practice here (#1490, #1584, #1445). There is deliberately no automatic translation to the replacements: `silence_threshold: 0.25` becoming `min_silence_duration_ms: 250` would move transcripts for anyone who had set it.

**The node's `vad_max_speech_duration_s` default moves 20 → 0.** It never reached the decoder, so "no limit" is what actually runs today; keeping `20` would make a plumbing fix change shipped transcripts as a side effect. `0` is the sentinel because faster-whisper's real default is `inf`, which JSON cannot express.

**`max_chunk_duration` is deliberately not exposed.** The first draft added it beside `chunk_duration`. @dsapandora caught that it can never fire: its `elif` runs only when the lower threshold did not, and any buffer long enough to reach it has already been flushed — `_flush_audio()` zeroes the counter besides. A second threshold means something only with a "cut here, it is quiet" test between the two, and this class has no silence detection; pause handling belongs to Whisper's VAD, as the class docstring has said since the initial commit. The field is gone and the dead branch with it. Whether seam-aware cutting is worth having at all is #1907, which leads with measurement rather than a fix.

**A new failure mode.** `vad_parameters={'nonsense': 1}` used to be discarded silently. It now reaches `VadOptions(**vad_parameters)` and raises, degrading that one item to a result carrying `error` rather than killing the batch. That is the correct trade — silently ignoring a caller's tuning is the bug being fixed — but it is a behaviour change.

**Transcript expectations drop `"Mr"`.** With profiles actually selecting their model, `small` and `tiny` write "Mr. Gorbachev" where `base`, `medium` and `large-v3` write "Mr Gorbachev". All five transcribe the line correctly; the mismatch was one period. Both `test` and `fulltest` now assert `"Gorbachev, tear down this wall."`, with a comment saying why.

**The generated README block is stale on purpose.** `nodes:docs-generate` only runs on `main`/`stage`/`develop`, so the four removed fields and the two new ones appear in the `ROCKETRIDE:GENERATED:PARAMS` block after this merges. Hand-editing inside the markers is forbidden.

## Identity: the #1750 convention

`WhisperLoader` never defined `_SERVER_PARAMS`, so keeping decode params out of model identity held only because the facade stopped sending them. An old client that still puts them in `loader_options` hashed to a different `model_id` and duplicated the weights through a rolling upgrade — the skew hole #1750 closed for GLiNER.

Separately, `load()` swallowed `**kwargs`, so `revision` changed the model id without reaching `WhisperModel()` — the default branch loaded under a different identity. Same bug `408ee1e8` fixed for GLiNER's `from_pretrained()`.

Both halves now follow that convention: `_SERVER_PARAMS` excludes the decode params from the hash, `load()` names them so they are absorbed rather than forwarded (forwarding would raise), and everything else in `**kwargs` reaches `WhisperModel()`.

## Verification

- 74 unit tests across `test_whisper_identity.py` and the new `nodes/test/audio_transcribe/test_decode_params.py`
- `nodes:test-full -k audio_transcribe` — 27 passed, all six profiles on their own models for the first time
- Baseline `nodes:test` captured before any edit; the `default` transcript is byte-identical after
- `docs:build`, ruff and gitleaks clean

Two guards were mutation-verified: reverting `vad_max_speech_duration_s` to `20` fails `test_empty_config_reproduces_todays_call`, and replacing the merge's `is None` filter with a truthiness check fails `test_zero_is_kept_on_every_other_vad_key`.

`IInstance` cannot be imported by the node test harness (relative imports), so its `Transcribe(...)` call is checked statically via AST. `transcribe.py` is loaded inside a synthetic package, so the flush itself is exercised for real.

## Skew

The saas forwarding channel has existed since saas #466 and already sweeps these names into `infer_kwargs`; it only ever dropped them because `inference()` did not declare them. New server + current saas works immediately. A saas PR follows with the submodule bump, the integration test, and a `pnpm-lock.yaml` refresh — the rebase range adds dependencies the saas lockfile does not know.

## Summary by CodeRabbit

* **New Features**
  * Added configurable Whisper decoding options, including beam size, VAD filtering, language, and word-level timestamps.
  * Added configurable audio chunking with unlimited-duration support.
  * Improved consistency of transcription settings across local and remote processing.

* **Bug Fixes**
  * Improved handling and defaults for transcription and VAD settings.
  * Updated transcription profiles to use model selection consistently.

* **Documentation**
  * Documented chunking, decoding, VAD options, defaults, and legacy configuration compatibility.

### CODE REVIEW

File: packages/ai/tests/ai/common/models/audio/test_whisper_identity.py [hunk 1]
```diff
@@ -157,3 +177,330 @@ def test_postprocess_still_falls_back_to_loaded_language():
...
+
+    for name in ('beam_size', 'vad_filter', 'vad_parameters', 'word_timestamps'):
+        assert name not in captured
+
+
+def test_load_forwards_genuine_loader_options(monkeypatch):
+    """Before this, revision changed the model id but never reached the constructor."""
+    captured = _load_with(monkeypatch, revision='abc123')
+
+    assert captured['revision'] == 'abc123'
```
[the remark this answers is not in the document]
Comment (asclearuc) [reply packages/ai/tests/ai/common/models/audio/test_whisper_identity.py:451]: Fixed in 049d6d10 — you were right on both halves.

`load()` calls `_ensure_dependencies()` and does `from ai.common.torch import torch` before it reaches `faster_whisper`, so stubbing only the latter left the resolver and the torch import live. It passed here because the engine has both installed, which is exactly the environment that hides it — and this module's docstring promises "No torch, no faster-whisper, no model download."

`_ensure_dependencies` is now patched on `WhisperLoader` and `ai.common.torch` is stubbed in `sys.modules` alongside `faster_whisper`. The patch is deliberately left at the default `raising=True`, so a misspelled name fails the test rather than silently doing nothing.

The `model_gpu_gb` patch is gone. You are right that it was a no-op: it is a module-level import at `whisper.py:30`, not a `WhisperLoader` member, so `raising=False` just attached a bogus attribute.

File: nodes/src/nodes/audio_transcribe/README.md [hunk 2]
```diff
@@ -76,12 +94,18 @@ Defaults to English (`en`). Change the `language` config value to transcribe oth
...
 | `transcribe.model` | `string` | **Model**<br/>The Whisper model to use for transcription | `"base"` |
 | `transcribe.profile` | `string` |  | `"default"` |
 | `transcribe.silence_threshold` | `number` | **Silence Threshold**<br/>The silence threshold to detect silence in speech (in seconds) | `0.25` |
+| `transcribe.vad_filter` | `boolean` | **VAD Filter**<br/>Use Whisper's built-in Silero VAD to drop non-speech before transcribing | `true` |
 | `transcribe.vad_level` | `number` | **VAD Level**<br/>The VAD level to use for silence detection (0-3) | `1` |
+| `transcribe.vad_max_speech_duration_s` | `number` | **VAD Maximum Speech (s)**<br/>Split speech chunks longer than this. 0 means no limit | `0` |
+| `transcribe.vad_min_silence_duration_ms` | `number` | **VAD Minimum Silence (ms)**<br/>Silence this long ends a speech chunk | `500` |
+| `transcribe.vad_speech_pad_ms` | `number` | **VAD Speech Padding (ms)**<br/>Padding added to each side of a detected speech chunk | `400` |
+| `transcribe.vad_threshold` | `number` | **VAD Speech Threshold**<br/>Silero speech probability above which audio counts as speech (0-1) | `0.5` |
```
[the remark this answers is not in the document]
Comment (asclearuc) [reply nodes/src/nodes/audio_transcribe/README.md:107]: Not applying this one — the premise does not hold.

Those rows inside the marker block were not hand-edited. They are generator output: `builder docs:build` ran while the checkout was still on `develop`, before this branch existed, so `nodes:docs-generate` was past its `main`/`stage`/`develop` guard and regenerated the block legitimately. (The same run also regenerated 29 unrelated node READMEs, which I reverted by path.) `git diff cf023e2b..HEAD` on this file shows only additions inside the markers — the six decode-param rows — and no edits to existing ones.

Reverting them would make the block strictly less accurate, not more. The four dead rows you flagged were already there on `develop` and are not touched by this PR, so a revert leaves them in place while also dropping six correct rows and still lacking `chunk_duration` / `max_chunk_duration`. The document stays self-contradictory either way, just in a worse direction.

The contradiction is real and unavoidable on a feature branch: the guard exists precisely because the branch name is baked into the generated source links, so regenerating here would be wrong for a different reason. As you note yourself, `nodes:docs-generate` runs as part of `nodes:build` on `develop` and will bring the whole block into line after merge. That is called out in the PR description so a reviewer does not read it as an oversight.

**Review by dsapandora — CHANGES_REQUESTED** [review]
`max_chunk_duration` doesn't reach anything yet.

`transcribe.py:170-173`:

```python
if self._total_samples >= self._chunk_samples:        # 60s
    self._flush_audio()
elif self._total_samples >= self._max_chunk_samples:  # 120s
    self._flush_audio()
```

The `elif` can't fire while `chunk_samples <= max_chunk_samples`, so the buffer always flushes at `chunk_duration` and the hard limit never runs. That leaves the new field as a knob that silently does nothing, which is the class of defect this PR is closing, and the README line "Hard limit before a chunk is flushed even without a pause" describes behaviour that isn't there. The "forced flush at 120 seconds" line further up has the same problem, though that one predates this PR.

The tests stay green with the flush path dead: `test_buffering_defaults_reproduce_transcribe_constants` compares the default to the constant, and `test_iinstance_passes_buffering_to_transcribe` is a static AST check. Neither observes a flush.

Two ways out, both fine by me — wire the hard limit to its own trigger, or drop `max_chunk_duration` from this PR and ship `chunk_duration` alone. Shipping the field as it stands is the one I'd avoid.

The rest reads well. The decode-param plumbing, the `_SERVER_PARAMS` exclusion so an old client can't split the weights, and the `is None` rather than truthiness merge of the VAD dict are all solid.

**Comment by asclearuc** [comment]
@dsapandora fixed in `e9648c78` — you were right, and the branch is rebased onto current `develop`.

`max_chunk_duration` is gone, together with the dead `elif`, `MAX_CHUNK_DURATION`, `MAX_CHUNK_SAMPLES` and `_max_chunk_samples`. The reasoning holds twice over: the `elif` cannot run because reaching it means already passing the lower threshold, and `_flush_audio()` zeroes the counter, so the buffer never approaches the higher value at all.

Two things I checked before deciding how to answer.

**It is not a regression.** The `elif` has been unreachable since `81977440`, the repository's root commit — as have `mode` vs `model` and all four dead config fields. The same file's docstring there already reads *"Whisper's built-in VAD handles pause detection and speech segmentation. This is much more efficient than external VAD processing."* So node-level silence cutting was removed deliberately, before this repository existed; `silence_threshold` (a pause threshold in seconds) and `vad_level` (webrtcvad's 0-3 aggressiveness) are its fossils. There is no prior behaviour to restore and none to pin, which is why this is not answered with a regression test.

**Your point about the tests is the important one.** None of the three observed a flush — two were AST checks, one read attributes. I verified the wiring as far as the constructor argument and stopped short of verifying that anyone reads it, which is exactly the gap that let an inert knob through. Replaced with a behavioural test: it feeds real PCM and asserts no flush before `chunk_duration`, exactly one at it carrying the right byte count, and none immediately after, because the counter resets. Mutation-checked — changing `>=` to `>` fails it. `transcribe.py` uses a relative import, so it now loads inside a synthetic package; sidestepping that is why I reached for AST in the first place.

The README lines are corrected too, including the intro sentence — which I had made worse by adding "both configurable" to a threshold that is never read.

On the other option you offered, wiring the hard limit to its own trigger: I opened #1907 instead of doing it here. A second threshold is meaningless without a "cut here, it is quiet" test between the two, so that is not plumbing but a new capability, and it moves every transcript longer than one chunk — the class of change this PR has refused elsewhere (word timestamps default to off; `min_seconds` 240/300 deliberately not honoured). The issue leads with measuring the seam cost rather than proposing a fix, because I have not measured it and the corpus here cannot: every clip is shorter than one chunk.

Verification after the change: 74 unit tests, `nodes:test-full` 27 passed across all six profiles, ruff and gitleaks clean.

**Review by dsapandora — APPROVED** [review]
Approving.

- `max_chunk_duration` is gone along with the dead `elif`, `MAX_CHUNK_DURATION`, `MAX_CHUNK_SAMPLES` and `_max_chunk_samples`. No references left.
- The replacement test observes a real flush instead of inspecting the constructor — flipping `>=` to `>` fails it.
- README no longer claims a 120 s forced flush.

#1907 is the right home for a pause-aware second threshold.
